### PR TITLE
Add table of contents generator/verify scripts

### PR DESCRIPTION
**Pull request by damemi**
This adds scripts to generate and verify that the README table of contents is updated
/kind feature

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
/cc @ingvagabund

File: hack/update-toc.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
+
+# forked from github.com/ekalinin/github-markdown-toc/gh-md-toc
+wget -O ${OS_OUTPUT_BINPATH}/gh-md-toc https://raw.githubusercontent.com/damemi/github-markdown-toc/hide-footer/gh-md-toc
```
Comment (seanmalloy) [inline hack/update-toc.sh:5]: Should the `gh-md-toc` script just be added to the descheduler repo instead of downloading it from another repo?

Comment (ingvagabund) [reply hack/update-toc.sh:5]: What about to duplicate https://github.com/kubernetes-sigs/metrics-server/blob/master/Makefile#L177-L189 ? Looks quite shorter.

Comment (damemi) [reply hack/update-toc.sh:5]: Oh sweet, I did not even know there was a k8s toc generator already. I'll update to use that instead.

We can vendor it like we do with our other generators so it doesn't have to be downloaded each time.

**Comment by damemi** [comment]
/close
keeping this PR as an example though

**Pull request closed without merging**